2025/incoming/the-gift-of-readability.pod

```diff
@@ -0,0 +1,450 @@
+Author: Branislav Zahradník <barney@cpan.org>
+Title: The Gift of Readability
+Topic: Source code
+
+=encoding utf8
+
+=head2 Night Shift at PDN Headquarters
+
+At the Present Delivery Network headquarters, December nights were always long.
+
+The elves — engineers by trade, magicians by necessity — sat beneath glowing dashboards
+tracking sleigh routes, Aurora Borealis hit ratios, and reindeer latency.
+
+All of it powered by Perl. Old Perl. New Perl.
+Perl written by elves long retired to toy-making or cloud-making.
+Perl written by skilled masters.
+Perl written by babies.
+
+She rubbed her eyes.
+
+Three hours. Three hours she'd been trying to understand this function.
+Three hours to fix what should have been a five-minute bug.
+
+The Present Delivery Network ran flawlessly — millions of gifts delivered every year with magical precision.
+But behind the scenes, the elves maintaining the codebase were drowning in their own cleverness.
+
+This code worked perfectly for ages, yet recently it stopped to feed Rudolph.
+And it is utterly incomprehensible for her.
+
+=begin code perl
+
+for my$r(@$rslts){push@${${$r->{feed}//next}->{$r->{reindeer}//next}},getPreferredReindeerFuel($r->{name}//next,$r->{prefs})if!!$r->{active}==!!1;}
+
+=end code
+
+=over
+
+=item * “Why,” she whispered to the empty office, “why do we do this to ourselves?”
+
+=item * "The Present Delivery Network was designed to be efficient."
+
+=item * "This code … less so."
+
+=back
+
+She thought about the astigmatic, colour-blind elf, squinting at compressed code through his magnifier.
+She thought about the foreigner elf, parsing English as their second language.
+She thought about the new elf starting after the holidays, terrified to touch anything.
+
+And so began the quiet refactoring ...
+
+=head2 The Invisible Barrier
+
+The Capability to Read is Not Given
+
+Readable code isn't just about aesthetics or dogma.
+
+It is about politeness.
+It is about inclusiveness.
```

Comment:
I particularly object to this notion. Every decision includes as well as excludes, but style, no matter how bad, does not actually exclude anyone. They have the same access as anyone else, the same chance to participate, and so on.